tool/gcs_sync.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2024 Google LLC
```

Comment:
I reckon this repeats https://github.com/google/oss-fuzz-gen/pull/940

Could you please chat with @sudhin05 to find the best way to divide the tasks?
Thanks : )